### PR TITLE
implement embedded-hal-async traits using maybe-async-cfg

### LINKED ISSUE #30: Support async
Hi @eldruin thanks for another great driver crate!

I'm looking to use this driver in an async context. Would you be in favour of adding driver implementations based on `embedded-hal-async` traits? If so, what would you suggest be the best way to go about it?

### PR DESCRIPTION
**Pull request by hdoordt**
Fixes #30 

To do:
- [x] Add usage example in README
- [x] Add usage example in rustdoc
- [x] Add example for microbit v2
- [x] Fix errors caused by `#[doc = include_str!("delay.md")]`'
- ~~[ ] Test async stuff~~

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -102,6 +102,7 @@
 //! ```
 
 #![deny(unsafe_code, missing_docs)]
+#![allow(async_fn_in_trait)]
```
Comment (hdoordt) [inline src/lib.rs:105]: I don't think we actually want this. However, fixing the warning is going to take writing a bunch of boilerplate, I'm afraid

**Review by eldruin — COMMENTED** [review]
Looks good to me so far, thank you!

**Comment by hdoordt** [comment]
@eldruin I'm having some trouble getting the tests to run for `async`. Here's my current attempt: <https://github.com/hdoordt/lsm303agr-rs/commit/91505ceeb56fa22f49c80b2ac922c5ebbeb5edfe>

Sadly, the attributes don't seem to work together all that well. Can you maybe have a look?

File: src/accel_mode_and_odr.rs [hunk 2]
```diff
@@ -15,8 +20,14 @@ where
...
-    #[doc = include_str!("delay.md")]
-    pub fn set_accel_mode_and_odr<D: DelayNs>(
+    /// The given `delay` is used to wait for the sensor to turn on or change modes,
+    /// according to the times specified in Table 14 and Table 15 in the [datasheet].
+    /// You can opt out of this by using a no-op delay implementation, see
+    /// [`embedded_hal_mock::delay::MockNoop`] for an example of such an
+    /// implementation.
+    /// [datasheet]: https://www.st.com/resource/en/datasheet/lsm303agr.pdf
+    /// [`embedded_hal_mock::delay::MockNoop`]: https://docs.rs/embedded-hal-mock/latest/embedded_hal_mock/delay/struct.MockNoop.html
+    pub async fn set_accel_mode_and_odr<D: DelayNs>(
```
Comment (hdoordt) [inline src/accel_mode_and_odr.rs:30]: I copied in the contents of `delay.md` here as well as in [magnetometer.rs:16-32](https://github.com/eldruin/lsm303agr-rs/pull/31/files#diff-655dfc5c14daec4123c6f7da6fd3026f88f91db687ed9cfc2a761bac52131ca6L16-R32)

Not too happy about the reduncancy, but the expansion of `maybe_async_cfg::maybe` does not work well with `#[doc = include_str!("<file>")]`

Comment (eldruin) [reply src/accel_mode_and_odr.rs:30]: ok no worries. Please delete the file since there are no users anymore.

**Comment by eldruin** [comment]
The tests themselves in https://github.com/hdoordt/lsm303agr-rs/commit/ded90ca154ded9da145b4c36bc31c163cf7e9759 work fine for me after adding a single `.await` in `accel_mode_and_odr.rs:297` and running `cargo test --features='async' --lib`.
However, the doctests and example cannot be compiled due to `linux-embedded-hal` not supporting async. That is expected and I would not try to do anything about it.
It is very interesting to be able to run async tests but honestly I would not have put the effort into it since the code is _so far_ the same.

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 <!-- next-header -->
 ## [Unreleased] - ReleaseDate
+- Add support for async behind the `async` feature flag
+- Update MSRV to 1.75.0
```
Comment (eldruin) [inline CHANGELOG.md:11]:
```suggestion

### Added
- Add support for async behind the `async` feature flag

### Changed
- Update MSRV to 1.75.0
```

File: memory.x [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# Linker script for the nRF52833 MCU on the microbit:v2
```
Comment (eldruin) [inline memory.x:1]:
```suggestion
# Linker script for the nRF52833 MCU on the microbit:v2 necessary for the example. Please disregard it otherwise.
```

**Review by eldruin — COMMENTED** [review]
The micro:bit example looks great :)

**Comment by hdoordt** [comment]
> The tests themselves in [hdoordt@ded90ca](https://github.com/hdoordt/lsm303agr-rs/commit/ded90ca154ded9da145b4c36bc31c163cf7e9759) work fine for me after adding a single `.await` in `accel_mode_and_odr.rs:297` and running `cargo test --features='async' --lib`. However, the doctests and example cannot be compiled due to `linux-embedded-hal` not supporting async. That is expected and I would not try to do anything about it. It is very interesting to be able to run async tests but honestly I would not have put the effort into it since the code is _so far_ the same.


Problem is that now the sync tests don't work anymore. I'll leave the async testing out if it's all the same to you

File: src/lib.rs (hunk 1 above)
Comment (eldruin) [reply src/lib.rs:109]: I agree and would leave it as-is.

**Review by eldruin — COMMENTED** [review]
Looks great. Just one trivial thing left: Could you fix the code formatting?

**Comment by hdoordt** [comment]
Done and done.

Could you create a release that includes this feature to crates.io? I'd like to use the driver in a training.

**Review by eldruin — APPROVED** [review]
Great work, thank you!

**Comment by eldruin** [comment]
I have released this in [version 1.1.0](https://github.com/eldruin/lsm303agr-rs/releases/tag/v1.1.0).
I have also removed the `microbit-example` feature, since we are not running the tests with `async` and `async` is a requirement for the example. Let me know if that is a problem for you.

**Pull request merged**